services/src/main/scala/cromwell/pipeline/service/HttpClient.scala

```diff
@@ -1,17 +1,74 @@
 package cromwell.pipeline.service
 
+/**
+ * This trait force implementations to provide REST operations using PlayJson, based on Reads & Writes
+ * @get,
+ * @post,
+ * @put methods have Response generic type, which is
+ * @ResponseBoby type of its Response
+ *
+ * Both ResponseBody (B) and Payload (P) are parametrized
+ * To use this client you have to have implicit Reads or Writes for your entities in your DTO
+ *
+ * @version 1.0
+ */
+import play.api.libs.json.{ Json, OFormat, OWrites, Reads, Writes }
+
 import scala.concurrent.{ ExecutionContext, Future }
 
+/**
+ * Trait with standard REST operations, with parametrized methods, where B is ResponseBody type and P is Payload type
+ * (bf reads as 'body formatter')
+ */
 trait HttpClient {
-  def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
-    implicit ec: ExecutionContext
-  ): Future[Response]
-  def post(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: String)(
-    implicit ec: ExecutionContext
-  ): Future[Response]
-  def put(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: String)(
-    implicit ec: ExecutionContext
-  ): Future[Response]
+
+  def get[B](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext,
+    f: Reads[B]
+  ): Future[Response[B]]
+  def post[B, P](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: P)(
+    implicit ec: ExecutionContext,
+    bf: Reads[B],
+    pf: Writes[P]
+  ): Future[Response[B]]
+  def put[B, P](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map(), payload: P)(
+    implicit ec: ExecutionContext,
+    bf: Reads[B],
+    pf: Writes[P]
+  ): Future[Response[B]]
 }
 
-case class Response(status: Int, body: String, headers: Map[String, Seq[String]])
+/**
+ * Contains response data with unmarshalled Json
+ *
+ * @param status equals Akka StatusCode
```

Comment:
I think you should remove any mention of Akka here. It is not a part of the HTTP client contract. It is an implementation detail that may change.

Reply:
akka mentions erased from ttait